### PR TITLE
Add support to navigate to / copy the venue address

### LINKED ISSUE #8: It would be great, if we can share the map in Venue section, so that people can open it in Google Maps.

### LINKED ISSUE #49: Make address of the venue for each session more actionable
Given that oftentimes users have to get to the venue in an unfamiliar city which involves using an external mapping app, it’d be great to allow copying address easily or prompting to open an installed maps

### PR DESCRIPTION
**Pull request by zats**
In a nutshell this just adds support of opening the venue address in an external maps app:
Fixes #49, fixes #8

img_75153be58343-1

### CODE REVIEW

File: trySwift/Maps/OpenInMapsProtocol.swift [hunk 1]
```diff
@@ -0,0 +1,70 @@
+//
```
Comment (BasThomas) [inline trySwift/Maps/OpenInMapsProtocol.swift]: Can you indent this file with four spaces?

File: trySwift/Maps/OpenInMapsProtocol.swift [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        } else {
+          assertionFailure("No address found for the placemark \(placemark)")
+          UIPasteboard.general.string = "\(placemark.coordinate.latitude), \(placemark.coordinate.longitude)"
+        }
+      })
+    }
+
+    alert.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: nil))
+
+    present(alert, animated: true, completion: nil)
```
Comment (BasThomas) [inline trySwift/Maps/OpenInMapsProtocol.swift]: Nit: you can omit the `completion` parameter.

File: trySwift/Maps/OpenInMapsProtocol.swift [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    })
+
+    if allowCopy {
+      alert.addAction(UIAlertAction(title: "Copy Address", style: .default) { _ in
+        if let postalAddress = placemark.postalAddress {
+          let formatter = CNPostalAddressFormatter()
+          let address = formatter.string(from: postalAddress)
+          UIPasteboard.general.string = address
+        } else {
+          assertionFailure("No address found for the placemark \(placemark)")
```
Comment (BasThomas) [inline trySwift/Maps/OpenInMapsProtocol.swift]: Nice!

File: trySwift/Maps/OpenInMapsProtocol.swift [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+          UIPasteboard.general.string = "\(placemark.coordinate.latitude), \(placemark.coordinate.longitude)"
+        }
+      })
+    }
+
+    alert.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: nil))
```
Comment (BasThomas) [inline trySwift/Maps/OpenInMapsProtocol.swift]: Nit: omit the `completion` parameter.

File: trySwift/Maps/OpenInMapsProtocol.swift [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+  }
+
+  private func openInAppleMaps(_ placemark: MKPlacemark) {
+    MKMapItem(placemark: placemark).openInMaps(launchOptions: nil)
+  }
+
+  private func openInGoogleMaps(_ placemark: MKPlacemark) {
+    UIApplication.shared.open(URL(string:
+      "comgooglemaps-x-callback://" +
```
Comment (BasThomas) [inline trySwift/Maps/OpenInMapsProtocol.swift]: What about using the direct URL scheme? https://developers.google.com/maps/documentation/urls/guide

Comment (zats) [reply trySwift/Maps/OpenInMapsProtocol.swift]: I think I wanted to add a pin and http scheme wouldn't allow for that… 🤔 but I ended up forgetting to do it anyway 🙂 
let me check

**Comment by BasThomas** [comment]
Thanks @zats!

**Pull request merged**